### PR TITLE
Use proper endpoints for unkeeping and bulk unkeeping.

### PR DESCRIPTION
**Pull request by yipingliao**
DO NOT MERGE - we're having a small problem on the server with the DELETE method expecting a payload, but that should not affect client-side code (too much).

@ahsu1230 @andrewconner Using the correct endpoints for unkeeping and bulk unkeeping while libraries are enabled.

https://app.asana.com/0/15523651812135/17258517427134

### CODE REVIEW

File: kifi-backend/modules/shoebox/angular/src/keep/keep.js [hunk 1]
```diff
@@ -285,19 +285,27 @@ angular.module('kifi')
         };
 
         scope.unkeep = function (keep) {
-          keepActionService.unkeepOne(keep).then(function () {
-            keep.makeUnkept();
+          if (scope.librariesEnabled) {
```
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/keep/keep.js:288]: Use different endpoints based on whether libraries are enabled. Looking forward to taking away all these hooks at the end of the month!

File: kifi-backend/modules/shoebox/angular/src/keeps/keepActionService.js [hunk 2]
```diff
@@ -162,7 +162,8 @@ angular.module('kifi')
       };
 
       $log.log('keepActionService.keepToLibrary()', data);
-      var url = env.xhrBase + '/libraries/' + libraryId + '/keeps';
+
+      var url = routeService.addKeepsToLibrary(libraryId);
```
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/keeps/keepActionService.js:166]: Unrelated to unkeeping; now using `routeService` to give us the route to adding keeps to a library.

File: kifi-backend/modules/shoebox/angular/src/keeps/keeps.js [hunk 3]
```diff
@@ -181,20 +184,42 @@ angular.module('kifi')
         scope.unkeep = function (keeps) {
           var selectedKeeps = scope.selection.getSelected(keeps);
 
-          keepActionService.unkeepMany(selectedKeeps).then(function () {
-            _.forEach(selectedKeeps, function (selectedKeep) {
-              selectedKeep.makeUnkept();
-            });
+          if (scope.librariesEnabled) {
```
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/keeps/keeps.js:187]: Another big if-conditional for libraries. This one is for bulk unkeeping.

**Comment by andrewconner** [comment]
Hm, what's this DELETE problem?

**Comment by ahsu1230** [comment]
I accidentally left a (parse.tolerantJson) in the `LibraryController.removeKeeps()` method
so the DELETE required a body, which it shouldn't.
Addressing the problem in this next PR #11366

**Pull request merged**